content/Simple dynamic settings for Django - Jorge Michelena.md

```diff
@@ -0,0 +1,101 @@
+
+
+### Editing periodic celery tasks schedules
+
+The project included a couple periodic tasks running with Celery and I wanted to be able to edit those tasks' schedules, in this case the solution was very straightforward thanks to [django-celery-beat](https://github.com/celery/django-celery-beat).
+
+Django-celery-beat is a library for Django that provides us with models for periodic tasks and models that help us to define and modify when and how those tasks will be executed; having a great degree of freedom. It can be done through code or the Django admin page, which results very helpful.
+
+Assuming you already have Celery integrated to your Django project all you have to do is install django-celery-beat and modify your settings file re-assigning or defining the `CELERY_BEAT_SCHEDULER` to be `'django_celery_beat.schedulers:DatabaseScheduler'`. After that all you have to do is run the `manage.py migrate` command and you are done.
+
+
+### Using a model to store configurations
+
+The fundamental idea is to define a model and use its fields to store the values that will be used as settings to determine the behavior of certain functionalities instead of using the *settings* file.
+Given the fact that we want the settings to be consistent wherever we read them from in the code it would be convenient to have only one instance of this model, in other words an implementation of the Singleton design pattern.
+Technically we won't be implementing a singleton, but for our purposes it will be the same.
+
+First we will define an abstract model that will be the basis for our settings model and that will provide the basic behavior:
+
+```python
+class Singleton(models.Model):
+
+    class Meta:
+        abstract = True
+
+    def save(self, *args, **kwargs):
+        self.pk = 1
+        super(Singleton, self).save(*args, **kwargs)
+
+    def delete(self, *args, **kwargs):
+        pass
+
+    @classmethod
+    def load(cls):
+        obj, _ = cls.objects.get_or_create(pk=1)
+        return obj
+```
+
+- The `save` method assigns a value of 1 to the primary key field of any instance of `Singleton` before saving it, overwriting any previously saved register in the database's singleton table.
+- The `load` method returns an instance of `Singleton` that represents the only existing record in the table (if there are no records then creates one).
+- Finally the `delete` method does nothing.
+
+As previously said, this implementation does not correspond to the Singleton pattern because it is possible for multiple instances of the model to exist at the same time, but it guarantees that in any moment at most one register associated to `Singleton` can exist in the database, wich gives us the consistency that we wanted.
```

Comment:
```suggestion
As previously stated, this implementation does not correspond to the Singleton pattern because it is possible for multiple instances of the model to exist at the same time, but it guarantees that in any moment at most one register associated to `Singleton` can exist in the database, which gives us the consistency that we wanted.
```